test/TritonGPU/coalesce.mlir

```diff
@@ -131,3 +131,13 @@ module attributes {"triton_gpu.num-warps" = 4 : i32, "triton_gpu.threads-per-war
     tt.return
   }
 }
+
+// -----
+
+
+module {
+  tt.func @int_min_does_not_underflow_in_analysis() -> i64 {
+    %int_min = arith.constant -9223372036854775808 : i64
+    tt.return %int_min : i64
+  }
```

Comment:
would this test fail without your patch?

Reply:
The test will fail if run with ASAN. It detects that underflow is happening because of an attempt to compute n - 1 which is below INT_MIN.

Reply:
it sounds like we were most likely returning the wrong value for int_min so we can have a test that doesn't rely on UBSAN and have a CHECK for the right divisibility?

Reply:
Done.